backend/src/main/java/ch/puzzle/okr/service/business/QuarterBusinessService.java

```diff
@@ -72,12 +85,34 @@ public YearMonth getCurrentYearMonth() {
         return YearMonth.now();
     }
 
+    public Map<Integer, Integer> generateQuarters() {
```

Comment:
Code mit + und - Operationen ist etwas schwer zu verstehen. Man könnte den vereinfachen, indem man alle Berechnungen "zero based" macht und erst beim Einfügen in die Map die Werte um 1 erhöht.


```java
Map<Integer, Integer> generateQuarters() {
    Map<Integer, Integer> quarters = new HashMap<>();
    int quarterIndexZeroBased = quarterStart - 1;
    for (int i = 0; i < 12; i++) {
        int monthIndexZeroBased = (i + quarterIndexZeroBased) % 12;
        int quarterZeroBased = i / 3;
        quarters.put(monthIndexZeroBased + 1, quarterZeroBased + 1);
    }
    return quarters;
}
```